utilities/PerformanceTesting/query.py

```diff
@@ -0,0 +1,279 @@
+
+            time_query_name = f'{g_stat_names[stat]} time taken by {timer} (microseconds)'
+
+            time_value = querier.query_and_get_value(time_query)
+            res[time_query_name] = time_value
+
+            try:
+                total_time[stat] += time_value['value']
+            except TypeError:
+                err_msg = "WARNING: The following query did not return a valid response\n"
+                sys.stderr.write(err_msg)
+                sys.stderr.write(time_query)
+
+    ave_json = {}
+    max_json = {}
+
+    ave_json['value'] = total_time['avg']
+    max_json['value'] = total_time['max']
+
+    avg_name = 'Total average time taken by collector timers (microseconds)'
+    max_name = 'Total time taken by collector timers in worst case (Sum of max times in microseconds)'
+
+    res[avg_name] = ave_json
+    res[max_name] = max_json
+
+    return res
+
+
+def get_collector_counters(querier):
+    """
+    Performs Prometheus queries relevant to Collector counters
+
+    :param querier: An object containing information
+    needed to query the Prometheus API
+
+    :return: Json containing information about Collector counters
+    """
+    res = {}
+    counters = ["net_conn_deltas", "net_conn_updates", "net_conn_inactive"]
+
+    for counter in counters:
+        for stat in g_stat_names:
+            query = f'{stat} by (job) (rox_collector_counters{{type="{counter}"}})'
+            query_name = f'{g_stat_names[stat]} of {counter} over pods'
+
+            res[query_name] = querier.query_and_get_value(query)
+
+    return res
+
+
+def get_sensor_network_flows(querier, query_window):
+    """
+    Performs Prometheus queries relevant to sensor network flows
+
+    :param querier: An object containing information
+    needed to query the Prometheus API
+    :param query_window: The window over which rates are computed by Prometheus
+
+    :return: Json containing information about sensor network flows
+    """
+    res = {}
+    for stat in g_stat_names:
+        for flow_type in ["incoming", "outgoing"]:
+            for protocol in ["L4_PROTOCOL_TCP", "L4_PROTOCOL_UDP"]:
+                metric = 'rox_sensor_network_flow_total_per_node'
+                query = f'{stat} by (job)  ({metric}{{Protocol="{protocol}",Type="{flow_type}"}})'
+                query_name = f'{g_stat_names[stat]} number of {flow_type} {protocol} network flows over pods'
+
+                res[query_name] = querier.query_and_get_value(query)
+
+    metrics = [
+                "rox_sensor_network_flow_host_connections_added",
+                "rox_sensor_network_flow_host_connections_removed",
+                "rox_sensor_network_flow_external_flows"
+            ]
+
+    for metric in metrics:
+        query = metric
+        rate_query = f'rate({metric}[{query_window}])'
+
+        query_name = metric
+        rate_query_name = 'Average {metric} per second'
```

Comment:
I think you might be missing an f here
```suggestion
        rate_query_name = f'Average {metric} per second'
```

Still, the output of this function falls into the same bag as my previous 2 comments.